cumulusci/tasks/bulkdata/load.py

```diff
@@ -120,7 +116,70 @@ def _init_options(self, kwargs):
             self.options.get("set_recently_viewed", True)
         )
 
+    def _init_dataset(self):
```

Comment:
I had to read the docs below to figure out what the intended control flow was in this method. A few comments explaining the order of fallback and how `org_shape_match_only` is intended to work would be helpful.